src/iuse_actor.cpp

```diff
@@ -973,27 +972,59 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
     if( g->m.flammable_items_at( pos ) ||
         g->m.has_flag( "FLAMMABLE", pos ) || g->m.has_flag( "FLAMMABLE_ASH", pos ) ||
         g->m.get_field_strength( pos, fd_web ) > 0 ) {
-        // Check for a brazier.
-        bool has_unactivated_brazier = false;
+        
+        bool has_unactivated_brazier = false; // Check for a brazier.
+        bool has_tinder_here = false;
         for( const auto &i : g->m.i_at( pos ) ) {
             if( i.typeId() == "brazier" ) {
                 has_unactivated_brazier = true;
             }
+            if( i.is_tinder() ) {
+                has_tinder_here = true;
+            }
         }
         if( has_unactivated_brazier &&
             !query_yn(_("There's a brazier there but you haven't set it up to contain the fire. Continue?")) ) {
             return false;
         }
-        return true;
+
+        inventory &inv = g->u.inv;
+        item &tinder = inv.find_item( inv.position_by_type( "tinder" ) );
+        if( has_tinder_here ) {
+            *density = 2;
+            return true;
+        } else if( inv.has_tools( "tinder", 1 ) ) {
+            int used_charges = tinder.type->stack_size / 2;
+            if( tinder.count_by_charges() && tinder.charges >= used_charges ) {
+                tinder.charges -= used_charges;
+                if( tinder.charges <= 0 ) {
+                    inv.remove_item( &tinder );
+                }
+
+                p.add_msg_if_player( m_info, string_format( ( "You use %d charges of tinder to light the fire." ), 
```

Comment:
Why are there `()` around the string literal. And it should be translated. Maybe only the `_` is missing? Same below.